internal/provider/harness.go

```diff
@@ -158,6 +160,40 @@ func (r *HarnessResource) ShouldSkip(ctx context.Context, req resource.CreateReq
 	return skip
 }
 
+// ParseMemoryResources parses memory resources for harnesses, returning a ContainerResourcesRequest object if parsing
+// is successful or an error if values are not parseable.
+func ParseMemoryResources(memoryResources ContainerMemoryResources) (*provider.ContainerResourcesRequest, error) {
```

Comment:
can we plumb through a `ParseCpuResources` in a similar way? that was the primary knob I was hoping to use for the pytorch tests 🙏 

if it warrants a follow up PR as to not make this overly messy that's fine too!

Reply:
I was planning to originally, but the Docker API is a little strange in that regard: https://docs.docker.com/config/containers/resource_constraints/#cpu

I'm really unsure which one of those would be the request vs limit, do you have any insights on this?

Reply:
my guess is its `--cpus` for the request, and `--cpu-quota` for the limits. either way, I approved this, will let you decide if you want to include it in a follow up or not

Reply:
Opened #98 to follow-up